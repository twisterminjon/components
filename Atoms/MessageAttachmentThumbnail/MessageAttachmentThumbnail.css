.messageattachmentthumbnail {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  width: auto;
  overflow: hidden;
  min-width: 180px;
  min-height: 75px;

  font-family: var(--font-message-page--message-bubble-family);
  font-size: var(--font-message-page--message-bubble-size);
  font-weight: var(--font-message-page--message-bubble-weight);

  cursor: pointer;
}

.messageattachmentthumbnail--received {
  margin: 8px;
}

.messageattachmentthumbnail img {
  max-width: 100%;
  object-fit: cover;
  height: 240px;
}

.messageattachmentthumbnail img:before {
  content: ' ';
  display: block;

  background: repeating-linear-gradient(
    45deg,
    var(--messagepage__received_bg),
    var(--messagepage__received_bg) 10px,
    var(--messagepage__sent_bg) 10px,
    var(--messagepage__sent_bg) 20px
  );

  position: absolute;
  top: -10px;
  left: 0;
  height: calc(100% + 10px);
  width: 100%;
}

.messageattachmentthumbnail img:after {
  content: 'ðŸ’” Image not found';
  display: block;

  font-size: 16px;
  text-align: center;
  padding: 10px 0px;
  color: var(--messagepage__received_fg);

  position: absolute;
  top: 5px;
  left: 0;
  width: 100%;
}

.messagethumbnail-timestamp {
  display: flex;
  align-items: center;
  position: absolute;
  bottom: 10px;
  line-height: 18px;
  left: 15px;
  padding: 0 7px;
  border-radius: 10px;
  background-color: rgba(92, 92, 92, 0.6);
}

.messagethumbnail-timestamp-text {
  font-size: 12px;
  color: var(--font_color_light);
  white-space: nowrap;
  pointer-events: none;
}

.messagethumbnail-loader {
  z-index: 0 !important;
  margin-left: 5px !important;
}
