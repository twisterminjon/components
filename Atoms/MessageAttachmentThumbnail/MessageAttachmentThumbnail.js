import React, { useMemo } from 'react'
import PropTypes from 'prop-types'
import { Loader } from 'semantic-ui-react'

import { DateFormat, ProjectDate } from '@shared/helpers'

import './MessageAttachmentThumbnail.css'

MessageAttachmentThumbnail.propTypes = {
  /** Content for the attachment */
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
  /** Timestamp for the message */
  timestamp: PropTypes.string.isRequired,
  /** Called after the attachment is clicked */
  onClick: PropTypes.func,
  /** Can show loading spinner near the time */
  sending: PropTypes.bool,
  /** ClassName for the wrapper */
  className: PropTypes.string,
  /** Style for the wrapper */
  style: PropTypes.object,
}

MessageAttachmentThumbnail.defaultProps = {
  onClick: () => {},
  className: '',
  style: {},
}

function MessageAttachmentThumbnail({ isOwn, timestamp, onClick, sending, children, className, style, ...rest }) {
  const receivedClass = !isOwn ? 'messageattachmentthumbnail--received' : ''

  const niceTime = useMemo(() => ProjectDate(timestamp).formatLocalUTC(DateFormat.LT), [timestamp])

  return (
    <div
      onClick={onClick}
      className={`messageattachmentthumbnail ${receivedClass} ${className}`.trim()}
      style={style}
      {...rest}>
      {children}
      <div className="messagethumbnail-timestamp">
        <span className="messagethumbnail-timestamp-text">{niceTime}</span>
        <Loader inline inverted active={sending} size="mini" className="messagethumbnail-loader" />
      </div>
    </div>
  )
}

export default MessageAttachmentThumbnail
