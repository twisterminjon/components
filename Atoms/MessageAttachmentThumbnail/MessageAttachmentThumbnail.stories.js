import React from 'react'

import { storiesOf } from '@storybook/react'
import { boolean, text, withKnobs } from '@storybook/addon-knobs'
import { action } from '@storybook/addon-actions'

import MessageAttachmentThumbnail from './MessageAttachmentThumbnail'

const stories = storiesOf('Provider/v1/Atoms', module)
stories.addDecorator(withKnobs)

stories.add('MessageAttachmentThumbnail', () => {
  const timestampLab = 'timestamp'
  const timestampDef = '2020-08-08T13:30:00'
  const timestampVal = text(timestampLab, timestampDef)

  const sendingLab = 'sending'
  const sendingDef = false
  const sendingVal = boolean(sendingLab, sendingDef)

  return (
    <div
      style={{
        position: 'relative',
        display: 'flex',
        flexDirection: 'column',
        width: 375,
        height: 628,
      }}>
      <MessageAttachmentThumbnail sending={sendingVal} timestamp={timestampVal} onClick={action('clicked')}>
        <img alt="attachment" src="https://www.fillmurray.com/800/1200" />
      </MessageAttachmentThumbnail>
    </div>
  )
})
