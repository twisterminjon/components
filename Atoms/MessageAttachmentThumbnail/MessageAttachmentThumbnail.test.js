import React from 'react'
import MessageAttachmentThumbnail from './MessageAttachmentThumbnail'

describe('MessageAttachmentThumbnail', () => {
  const mockFun = jest.fn()

  it('matches the snapshot', () => {
    const wrapper = window.shallow(
      <MessageAttachmentThumbnail onClick={mockFun} timestamp="2020-08-08T13:30:00">
        <img alt="attachment" src="https://www.fillmurray.com/800/1200" />
      </MessageAttachmentThumbnail>
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('matches the snapshot for sending', () => {
    const wrapper = window.shallow(
      <MessageAttachmentThumbnail isOwn={true} sending={true} onClick={mockFun} timestamp="2020-08-08T13:30:00">
        <img alt="attachment" src="https://www.fillmurray.com/800/1200" />
      </MessageAttachmentThumbnail>
    )

    expect(wrapper).toMatchSnapshot()
  })
})
