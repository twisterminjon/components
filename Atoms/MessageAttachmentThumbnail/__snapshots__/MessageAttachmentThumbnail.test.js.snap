// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageAttachmentThumbnail matches the snapshot 1`] = `
<div
  className="messageattachmentthumbnail messageattachmentthumbnail--received"
  onClick={[MockFunction]}
  style={Object {}}
>
  <img
    alt="attachment"
    src="https://www.fillmurray.com/800/1200"
  />
  <div
    className="messagethumbnail-timestamp"
  >
    <span
      className="messagethumbnail-timestamp-text"
    >
      1:30 PM
    </span>
    <Loader
      className="messagethumbnail-loader"
      inline={true}
      inverted={true}
      size="mini"
    />
  </div>
</div>
`;

exports[`MessageAttachmentThumbnail matches the snapshot for sending 1`] = `
<div
  className="messageattachmentthumbnail"
  onClick={[MockFunction]}
  style={Object {}}
>
  <img
    alt="attachment"
    src="https://www.fillmurray.com/800/1200"
  />
  <div
    className="messagethumbnail-timestamp"
  >
    <span
      className="messagethumbnail-timestamp-text"
    >
      1:30 PM
    </span>
    <Loader
      active={true}
      className="messagethumbnail-loader"
      inline={true}
      inverted={true}
      size="mini"
    />
  </div>
</div>
`;
